src/__init__.py

```diff
@@ -275,11 +275,16 @@ def generate_basic(self):
             self.multiworld.itempool.remove(item_to_place)
 
         after_generate_basic(self, self.multiworld, self.player)
-
+        
+        if not hasattr(self, 'manual_item_counts'): #Add a dict so this is only computed once per player, otherwise for a 2 player generation it add 25 seconds of proccess
```

Comment:
Could just add this as a class prop around line 69 and default it to {}. Also, we don't need to have "manual" in the var name, since it's implied, and the scope is implied to be the current World object.